### PR TITLE
ZIO Test: Add Documentation for MockClock

### LINKED ISSUE #1443: ZIO Test: Improve Documentation of MockClock
`MockClock` provides significantly enhanced functionality for testing code that involves the passage of time, but it requires users to adjust the time using the `adjust` and `setTime` methods instead of just waiting for "wall time" to pass. We need better documentation so users understand how to take advantage of this in writing tests. This recently came up in #1329.

I will take the ticket.

### PR DESCRIPTION
**Pull request by adamgfraser**
Resolves #1443.

### CODE REVIEW

File: test/shared/src/main/scala/zio/test/mock/MockClock.scala [hunk 1]
```diff
@@ -26,6 +26,47 @@ import zio.internal.{ Scheduler => IScheduler }
 import zio.internal.Scheduler.CancelToken
 import zio.scheduler.Scheduler
 
+/**
+ * `MockClock` makes it easy to deterministially and efficiently test effects
```
Comment (ghostdogpr) [inline test/shared/src/main/scala/zio/test/mock/MockClock.scala]: `deterministially` => `deterministically`

Comment (adamgfraser) [reply test/shared/src/main/scala/zio/test/mock/MockClock.scala]: Thanks!

File: test/shared/src/main/scala/zio/test/mock/MockClock.scala [hunk 2]
```diff
@@ -26,6 +26,47 @@ import zio.internal.{ Scheduler => IScheduler }
...
+ * line below. Thus, a useful pattern when using `MockClock` is to fork the
+ * effect being tested, then adjust the clock to the desired time, and finally
+ * verify that the expected effects have been performed.
+ *
+ * One other thing to keep in mind is that `sleep` and related combinators
+ * schedule events to occur at an absolute point in time (e.g. 10 seconds from
+ * the `0` time), not relative time (e.g. 10 seconds from the current set clock
+ * time). This is necessary to avoid race conditions between scheduling effects
+ * and adjusting the time but means it sometimes may be necessary to use
+ * `adjustTime` to reset the time between effects you are testing to achieve
```
Comment (jozic) [inline test/shared/src/main/scala/zio/test/mock/MockClock.scala]: would be great to have an example here showing this use case

Comment (adamgfraser) [reply test/shared/src/main/scala/zio/test/mock/MockClock.scala]: Good suggestion. I will work on something and submit a follow up PR.

**Pull request merged**